src/runtime/hexagon_dma_device_shim.h

```diff
@@ -0,0 +1,207 @@
+/*!
+ * This file has the necessary structures and APIs for Initiating, executing and finishing a Hexagon DMA transfer
+ * The functions in this header file are the interfacing functions between Halide runtime and the Hexagon DMA driver
+ * The functions in this file lead to the hexagon DMA driver calls in case of availability of DMA driver and hexagon DMA tools
+ * In case of un availabilty of hexagon SDK tools and DMA drivers these function while mimic DMA with dummy transfers
+ */
+
+#ifndef _DMA_DEVICE_SHIM_H_
+#define _DMA_DEVICE_SHIM_H_
+
+#include "mini_qurt.h"
+#include "hexagon_mini_dma.h"
+using namespace Halide::Runtime::Internal::Qurt;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+__inline static int ceiling(int num, int div) {
+    return ((num + div -1)/(div));
+}
+
+__inline static int align(int x,int a) {
+    return (ceiling(x, a) * a);
+}
+
+/*!
+ * Params needed for Prepare for transfer
+ */
+typedef struct {
+    void* handle;
+    uintptr_t host_address;
+    int frame_width;
+    int frame_height;
+    int frame_stride;
+    int roi_width;
+    int roi_height;
+    int luma_stride;
+    int chroma_stride;
+    bool read;
+    t_eDmaFmt chroma_type;
+    t_eDmaFmt luma_type;
+    int ncomponents;
+    bool padding;
+    bool is_ubwc;
+    uintptr_t desc_address;
+    int desc_size;
+} t_dma_prepare_params;
+
+/*!
+ * Params needed to move data
+ */
+typedef struct {
+    void* handle;
+    int xoffset;
+    int yoffset;
+    int roi_width;
+    int roi_height;
+    int offset;
+    int l2_chroma_offset;
+    int ncomponents;
+    uintptr_t ping_buffer;
+} t_dma_move_params;
+
+typedef struct {
+    int u16W;
+    int u16H;
+} t_dma_pix_align_info;
+
+/*!
```

Comment:
Are these public API functions? If so, they should use doxygen comments (can't tell if this is doxygen style or not).